orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RestartStageHandler.kt

```diff
@@ -58,13 +60,40 @@ class RestartStageHandler(
         if (topStage.status.isComplete) {
           topStage.addRestartDetails(message.user)
           topStage.reset()
+          restartParentPipelineIfNeeded(message, topStage)
           repository.updateStatus(topStage.execution.type, topStage.execution.id, RUNNING)
           queue.push(StartStage(startMessage))
         }
       }
     }
   }
 
+  private fun restartParentPipelineIfNeeded(message: RestartStage, topStage: Stage) {
+    if (topStage.execution.trigger !is PipelineTrigger) {
+      return
+    }
+
+    val trigger = topStage.execution.trigger as PipelineTrigger
+
+    if (trigger.parentExecution.status != TERMINAL) {
```

Comment:
I used `TERMINAL` here, but I could have used something else like `STOPPED`. Not sure which is best

Reply:
Why only restart in the `TERMINAL` case? I think it's ok to "restart" so long as the parent is not RUNNING, no?
In that case, you can
```suggestion
    if (trigger.parentExecution.status.isComplete()) {
```